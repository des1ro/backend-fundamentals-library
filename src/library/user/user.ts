import { randomUUID } from "crypto";

export class User {
  private readonly uuid: string = randomUUID();
  private credit: number = 0;
  constructor(private readonly name: string) {}
  getName(): string {
    return this.name;
  }
  getUuid(): string {
    return this.uuid;
  }
  resetCredit() {
    this.credit = 0; // Czy tu powinienem wrzucić credit do construktora, (private readonly credit) i chcąc zmiany tworzyć new User?
  }
  getCredit() {
    return this.credit;
  }
  setCredit(credit: number): void {
    this.credit = credit;
  }
}
