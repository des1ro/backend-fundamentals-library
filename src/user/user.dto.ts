import { randomUUID } from "crypto";

export class User {
  readonly uuid: string = randomUUID();
  constructor(
    readonly name: string,
    private dueDate = 0,
    private penaltyPoints = 0
  ) {}
  getDueDate(): number {
    return this.dueDate;
  }
  getPenaltyPoints(): number {
    return this.penaltyPoints;
  }
  setDueDate(dueDate: number): void {
    this.dueDate = dueDate;
  }
  setPenaltyPoints(penaltyPoints: number): void {
    this.penaltyPoints = penaltyPoints;
  }
}
