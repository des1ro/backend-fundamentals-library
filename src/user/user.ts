import { randomUUID } from "crypto";
import { isAfter } from "date-fns";
export class User {
  readonly uuid: string = randomUUID();
  private dueDate?: number;
  private penaltyPoints = 0;
  constructor(readonly name: string) {}
  getPenaltyPoints() {
    return this.penaltyPoints;
  }
  addPenaltyPoints(penaltyPoints: number): void {
    this.penaltyPoints += penaltyPoints;
  }
  subtractPenaltyPoints(penaltyPoints: number): void {
    this.penaltyPoints -= penaltyPoints;
  }
  getDueDate() {
    return this.dueDate || 0;
  }
  setDueDate(date: number): void {
    this.dueDate = date;
  }
  isBanned(): boolean {
    return isAfter(this.getDueDate(), new Date());
  }
}
