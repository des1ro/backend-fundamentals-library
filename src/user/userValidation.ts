import { isAfter } from "date-fns";
import { User } from "./user";

export class UserValidation {
  constructor(private readonly user: User) {}
  isUserBanned() {
    return isAfter(this.user.getDueDate(), new Date());
  }
}
