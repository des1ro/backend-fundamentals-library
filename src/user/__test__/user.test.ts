import { User } from "../user.dto";
jest.mock("crypto", () => ({
  randomUUID: jest.fn().mockReturnValue("test-uuid"),
}));
describe("Book test suite", () => {
  it("Should create user object correctly", () => {
    //Given
    const testName = "Test name";
    const testDueDate = 1647959027055;
    const testPenaltyPoints = 20;
    //When
    const objectUnderTest = new User(testName, testDueDate, testPenaltyPoints);
    //Then
    expect(objectUnderTest).toMatchSnapshot();
  });
});
