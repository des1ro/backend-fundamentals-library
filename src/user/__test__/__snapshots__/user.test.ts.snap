// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book test suite Should create user object correctly 1`] = `
User {
  "dueDate": 1647959027055,
  "name": "Test name",
  "penaltyPoints": 20,
  "uuid": "test-uuid",
}
`;
